Nightly failure: sanity/transactions.py

Fails occasionally. Example runs:

http://nightly.neartest.com/run/2cef4105736ab969470df2751e78fd3f0fc0a81d_200428_102644
http://nightly.neartest.com/run/2cef4105736ab969470df2751e78fd3f0fc0a81d_200428_072213

Failure:

```
Traceback (most recent call last):
  File "tests/sanity/transactions.py", line 66, in <module>
    assert False, "Balances before: %s\nExpected balances: %s\nCurrent balances: %s\nSent at height: %s, cur height: %s\n" % (last_balances, ctx.expected_balances, ctx.get_balances(), sent_height, height)
AssertionError: Balances before: [1000000000000000000000000000000241, 1000000000000000000000000000001066, 999999999999999999999999999998829, 999999999999999999999999999999864]
Expected balances: [1000000000000000000000000000000974, 1000000000000000000000000000000619, 999999999999999999999999999998712, 999999999999999999999999999999695]
Current balances: [1000000000000000000000000000000974, 1000000000000000000000000000000619, 999999999999999999999999999998712, 999999999999999999999999999999695]
Sent at height: 59, cur height: 72

```

Sent at height is very often == 9 (mod 10), though in at least one run it was == 0 (mod 10)

--- a/chain/client/src/client.rs
+++ b/chain/client/src/client.rs
@@ -1175,12 +1175,15 @@ impl Client {
                     shard_id,
                     is_forwarded
                 );
-                let new_transaction = self.shards_mgr.insert_transaction(shard_id, tx.clone());
+                self.shards_mgr.insert_transaction(shard_id, tx.clone());
 
+                // Active validator:
+                //   possibly forward to next epoch validators
+                // Not active validator:
+                //   forward to current epoch validators,
+                //   possibly forward to next epoch validators
                 if active_validator {
-                    // Don't forward to next epoch validators if we've already seen the tx.
-                    // This is to prevent forwarding loops.
-                    if new_transaction && !is_forwarded {
+                    if !is_forwarded {
                         self.possibly_forward_tx_to_next_epoch(tx)?;
                     }
                     Ok(NetworkClientResponses::ValidTx)
@@ -1205,7 +1208,7 @@ impl Client {
         }
     }
 
-    /// Determine if I am a validator in next few blocks for specified shard.
+    /// Determine if I am a validator in next few blocks for specified shard, assuming epoch doesn't change.
     fn active_validator(&self, shard_id: ShardId) -> Result<bool, Error> {
         let head = self.chain.head()?;
 
@@ -1221,12 +1224,7 @@ impl Client {
                 head.height + i,
                 shard_id,
             )?;
-            let next_epoch_chunk_producer = self.runtime_adapter.get_chunk_producer(
-                &head.next_epoch_id,
-                head.height + i,
-                shard_id,
-            )?;
-            if &chunk_producer == account_id || &next_epoch_chunk_producer == account_id {
+            if &chunk_producer == account_id {
                 return Ok(true);
             }
         }

